ArmPlatformPkg/Library/PL011SerialPortLib/PL011SerialPortLib.c

```diff
@@ -34,22 +34,33 @@ SerialPortInitialize (
   EFI_PARITY_TYPE     Parity;
   UINT8               DataBits;
   EFI_STOP_BITS_TYPE  StopBits;
+  EFI_STATUS          Status;
+  UINT8               Scratch;
 
   BaudRate         = FixedPcdGet64 (PcdUartDefaultBaudRate);
   ReceiveFifoDepth = 0;         // Use default FIFO depth
   Parity           = (EFI_PARITY_TYPE)FixedPcdGet8 (PcdUartDefaultParity);
   DataBits         = FixedPcdGet8 (PcdUartDefaultDataBits);
   StopBits         = (EFI_STOP_BITS_TYPE)FixedPcdGet8 (PcdUartDefaultStopBits);
 
-  return PL011UartInitializePort (
-           (UINTN)PcdGet64 (PcdSerialRegisterBase),
-           PL011UartClockGetFreq (),
-           &BaudRate,
-           &ReceiveFifoDepth,
-           &Parity,
-           &DataBits,
-           &StopBits
-           );
+  Status = PL011UartInitializePort (
+             (UINTN)PcdGet64 (PcdSerialRegisterBase),
+             PL011UartClockGetFreq (),
+             &BaudRate,
+             &ReceiveFifoDepth,
+             &Parity,
+             &DataBits,
+             &StopBits
+             );
+
+  if (EFI_ERROR (Status)) {
+    return Status;
+  }
+
+  while (SerialPortRead (&Scratch, sizeof (Scratch)) != 0) {
+  }

```

Comment:
So without the other change, this is now a blocking read of 1 byte. Shouldn't you be doing something like

```
while (SerialPortPoll ()) {
  SerialPortRead();
}
```

instead?

Reply:
Yes, of course it should, otherwise it would hang there.

Thanks, its updated